core/templates/pages/learner-dashboard-page/learner-dashboard-page.component.spec.ts

```diff
@@ -638,253 +1636,317 @@ describe('Learner dashboard page', function() {
-        var description = 'Description';
-        ctrl.showSuggestionModal(newContent, oldContent, description);
-
-        expect(SuggestionModalForLearnerDashboardService.showSuggestionModal)
+      () => {
+        spyOn(suggestionModalForLearnerDashboardService, 'showSuggestionModal')
+          .and.returnValue(null);
+        let newContent = 'New content';
+        let oldContent = 'Old content';
+        let description = 'Description';
+        component.showSuggestionModal(newContent, oldContent, description);
+
+        expect(suggestionModalForLearnerDashboardService.showSuggestionModal)
           .toHaveBeenCalledWith('edit_exploration_state_content', {
             newContent: newContent,
             oldContent: oldContent,
             description: description
           });
       });
 
-    it('should open remove activity modal when removing activity', function() {
-      spyOn($uibModal, 'open').and.callThrough();
+    it('should open remove activity modal when removing activity', () => {
+      let sectionNameI18nId = 'I18N_LEARNER_DASHBOARD_INCOMPLETE_SECTION';
+      let subsectionName = 'I18N_DASHBOARD_EXPLORATIONS';
+      let activity = LearnerExplorationSummary.createFromBackendDict(
+        learnerDashboardData.exploration_playlist[1]);
+      component.openRemoveActivityModal(
+        sectionNameI18nId, subsectionName, activity);
+    });
 
-      var sectionNameI18nId = 'I18N_LEARNER_DASHBOARD_INCOMPLETE_SECTION';
-      var subsectionName = 'I18N_DASHBOARD_EXPLORATIONS';
-      var activity = {};
-      ctrl.openRemoveActivityModal(sectionNameI18nId, subsectionName, activity);
+    it('should not remove an incomplete exploration if it is not present',
+      fakeAsync(() => {
+        expect(component.incompleteExplorationsList.length).toBe(12);
 
-      expect($uibModal.open).toHaveBeenCalled();
-    });
+        let sectionNameI18nId = 'I18N_LEARNER_DASHBOARD_INCOMPLETE_SECTION';
+        let subsectionName = 'I18N_DASHBOARD_EXPLORATIONS';
+        // Get exploration with id 100.
+        let activity = LearnerExplorationSummary.createFromBackendDict(
+          // TODO(#10875): Fix type mismatch.
+          {
+            id: '100'
+          } as unknown as LearnerExplorationSummaryBackendDict);
+        component.openRemoveActivityModal(
+          sectionNameI18nId, subsectionName, activity);
+
+        tick();
+        fixture.detectChanges();
+
+        expect(component.incompleteExplorationsList.length).toBe(12);
+      }));
+
+    it('should not remove an exploration from exploration playlist' +
+      ' if it is not present', fakeAsync(() => {
+      expect(component.explorationPlaylist.length).toBe(10);
+
+      let sectionNameI18nId = 'I18N_LEARNER_DASHBOARD_PLAYLIST_SECTION';
+      let subsectionName = 'I18N_DASHBOARD_EXPLORATIONS';
+      // Get exploration with id 100.
+      let activity = LearnerExplorationSummary.createFromBackendDict(
+        // TODO(#10875): Fix type mismatch.
```

Comment:
Is this possible to fix now?

Reply:
If its hard, lets defer to a future PR, just wondering what blocks it

Reply:
I'm not sure, I couldn't find any type mismatch. Deferring to @darksun27.